lib/runtime-abi/src/vfs/file_like.rs

```diff
@@ -0,0 +1,23 @@
+use std::io;
+
+pub type Fd = isize;
+
+#[derive(Debug)]
+pub struct Metadata {
+    pub len: usize,
+    pub is_file: bool,
+}
+
+pub trait FileLike: std::io::Seek {
+    // get metadata
+    fn metadata(&self) -> Result<Metadata, failure::Error>;
+
+    // write
+    fn write_file(&mut self, buf: &[u8], offset: usize) -> Result<usize, io::Error>;
```

Comment:
We can get away with implementing Seek and not requiring the read and write to take offsets.

So, `FileLike` could be this:

```rust
pub trait FileLike {
    fn metadata(&self) -> Result<Metadata, failure::Error>;
    fn write(&mut self, buf: &[u8], offset: usize) -> Result<usize, io::Error>;
    fn read(&mut self, buf: &mut [u8], offset: usize) -> Result<usize, io::Error>;
    fn set_file_len(&mut self, len: usize) -> Result<(), failture::Error>;
}
```
or
```rust
pub trait FileLike: io::Seek {
    fn metadata(&self) -> Result<Metadata, failure::Error>;
    fn write(&mut self, buf: &[u8]) -> Result<usize, io::Error>;
    fn read(&mut self, buf: &mut [u8]) -> Result<usize, io::Error>;
    fn set_file_len(&mut self, len: usize) -> Result<(), failture::Error>;
}
```